src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,8 +32,38 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        setAddressElements(address);
+
     }
 
+    /**
+     * Creates an object for Block, Street, Unit, and PostalCode class depending on the address input
+     * @param address
+     */
+
+    private void setAddressElements(String address) {
+        String[] addressList = address.split(",");
+        if (addressList.length == 1) {
```

Comment:
It is good that you try to consider the cases where parts of the address can be missing. However, you still have to make assumptions about the missing parts to make sure your code work. For example, what if only street is missing, the length will be 3 but your parsing will be wrong. Therefore, in this case, just parse the input as if all the fields will be provided. This will greatly shortens your code as well.